ot/mpz-ot/src/kos/sender.rs

```diff
@@ -233,6 +196,64 @@ where
     type Msg = Message<BaseOT::Msg>;
 }
 
+#[async_trait]
+impl<BaseOT> OTSetup for Sender<BaseOT>
+where
+    BaseOT: OTSetup + OTReceiver<bool, Block> + Send,
+{
+    async fn setup<
+        Si: IoSink<Message<BaseOT::Msg>> + Send + Unpin,
+        St: IoStream<Message<BaseOT::Msg>> + Send + Unpin,
+    >(
+        &mut self,
+        sink: &mut Si,
+        stream: &mut St,
+    ) -> Result<(), OTError> {
+        if self.state.is_extension() {
+            return Ok(());
+        }
+
```

Comment:
@sinui0  , I dont think your last commit addressed this.
in `receiver.rs` as soon as we enter `setup` we set the error state:
```rust
let ext_receiver = self
            .state
            .replace(State::Error)
            .into_initialized()
            .map_err(ReceiverError::from)?;
```

here we should similarly set an error state immediately upon entering `setup` in case if an error is returned during cointoss or base OT setup.